cores/esp8266/core_esp8266_main.cpp

```diff
@@ -400,79 +400,230 @@ extern "C" void __disableWiFiAtBootTime (void)
+        // 0x5000", "RF_CAL =  size - 0x4000", and "SYSTEM_PARAMETER = size -
+        // 0x3000" are positioned in the last five sectors of flash memory.
+        // PHY_INIT_DATA is special. It is a one time read of 128 bytes of data
+        // that is provided by a spoofed flash read.
+        #if FLASH_MAP_SUPPORT
+        flash_size = __flashdesc[__flashindex].flash_size_kb * 1024u;
+        #else
+        // flashchip->chip_size is updated by the SDK. The size is based on the
+        // value patched into the .bin header by esptool.
+        // system_get_flash_size_map() returns that patched value.
+        flash_size = flashchip->chip_size;
+        #endif
+
+        // For all configurations, place RF_CAL and system_parameter in the
+        // last 4 sectors of the flash chip.
+        rf_cal = flash_size - 0x4000u;
+        system_parameter = flash_size - 0x3000u;
+
+        // The system_partition_table_regist will not allow partitions to
+        // overlap. EEPROM_start is a good choice for phy_data overlay. The SDK
+        // does not need to know about EEPROM_start. So we can omit it from the
+        // table. The real EEPROM access is after user_init() begins long after
+        // the PHY_DATA read. So it should be safe from conflicts.
+        phy_data = EEPROM_start - 0x40200000u;
+
+        // For SDKs 3.0 builds, "sdk3_begin_phy_data_spoof and
+        // user_rf_cal_sector_set" starts and stops the spoofing logic in
+        // `core_esp8266_phy.cpp`.
+        extern void sdk3_begin_phy_data_spoof();
+        sdk3_begin_phy_data_spoof();
+
+        ld_config_chip_size = phy_data + 4096 * 5;
+
+        // -DALLOW_SMALL_FLASH_SIZE=1
+        // Allows for small flash-size builds targeted for multiple devices,
+        // commonly IoT, of varying flash sizes.
+        #if !defined(FLASH_MAP_SUPPORT) & !defined(ALLOW_SMALL_FLASH_SIZE)
+        // Note, system_partition_table_regist will only catch when the build
+        // flash size value set by the Arduino IDE Tools menu is larger than
+        // the firmware image value detected and updated on the fly by esptool.
+        if (flashchip->chip_size != ld_config_chip_size) {
+            // Stop to avoid possible stored flash data corruption. This
+            // mismatch will not occur with flash size selection "Mapping
+            // defined by Hardware and Sketch".
+            chip_sz_str = PSTR("Flash size mismatch, check that the build setting matches the device.\n");
+            continue;
+        }
+        #elif defined(ALLOW_SMALL_FLASH_SIZE) && !defined(FLASH_MAP_SUPPORT)
+        // Note, while EEPROM is confined to a smaller flash size, we are still
+        // placing RF_CAL and SYSTEM_PARAMETER at the end of flash. To prevent
+        // this, esptool or its equal needs to not update the flash size in the
+        // .bin image.
+        #endif
+
+        #if FLASH_MAP_SUPPORT & defined(DEBUG_ESP_PORT)
```

Comment:
```suggestion
        #if FLASH_MAP_SUPPORT && defined(DEBUG_ESP_PORT)
```
(would also suggest moving all `#...` to the beginning of line? things are pretty dense here)

Reply:
I have gone back and forth on this. I kept getting lost with all the `#if`s in column one. The density of the `#if`s is the reason I started indenting the innermost `#if`s to reduce the visual disruption to a block of "C" code. When nested the outer `#if` are at the beginning of the line.  I don't know if doing this is a style violation.